modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java

```diff
@@ -0,0 +1,1144 @@
+
+		rollUpQuery = "REPLACE INTO experiment_rollup "
+				+ "(experiment_id, day, cumulative, bucket_label, action, "
+				+ "impression_count, impression_user_count, "
+				+ "action_count, action_user_count,context) "
+				+ "values(unhex('" + experimentId + "'),'" + day + "',1,'"
+				+ BLUE + "','',"
+				+ (countImpressionBlueProdTomorrowCumulative[0]) + ","
+				+ (countImpressionBlueProdTomorrowCumulative[1]) + ","
+				+ (countActionBlueProdTomorrowCumulative[0]) + ","
+				+ (countActionBlueProdTomorrowCumulative[1]) + ",'" + PROD
+				+ "')";
+
+		insertIntoRollUp(rollUpQuery);
+
+		rollUpQuery = "REPLACE INTO experiment_rollup "
+				+ "(experiment_id, day, cumulative, bucket_label, action, "
+				+ "impression_count, impression_user_count, "
+				+ "action_count, action_user_count,context) "
+				+ "values(unhex('" + experimentId + "'),'" + day + "',1,'"
+				+ RED + "',''," + (countImpressionRedProdTomorrowCumulative[0])
+				+ "," + (countImpressionRedProdTomorrowCumulative[1]) + ","
+				+ (countActionRedProdTomorrowCumulative[0]) + ","
+				+ (countActionRedProdTomorrowCumulative[1]) + ",'" + PROD
+				+ "')";
+
+		insertIntoRollUp(rollUpQuery);
+
+		String queryActionByBucketActionContextCumulative = "SELECT bucket_label, action, context, COUNT(user_id) as user_count, "
+				+ " COUNT(DISTINCT user_id) as distinct_user_count "
+				+ " FROM event_action where hex(experiment_id) = '"
+				+ experimentId
+				+ "' and timestamp <= '"
+				+ this.tomorrow
+				+ "' GROUP BY bucket_label, action ";
+
+		try (Statement statementActionByBucketActionContextCumulative = connection
+				.createStatement();
+				ResultSet resultActionByBucketActionContextCumulative = statementActionByBucketActionContextCumulative
+						.executeQuery(queryActionByBucketActionContextCumulative)) {
+
+			while (resultActionByBucketActionContextCumulative.next()) {
+				String bucket = resultActionByBucketActionContextCumulative
+						.getString("bucket_label");
+				String action = resultActionByBucketActionContextCumulative
+						.getString("action");
+				String context = resultActionByBucketActionContextCumulative
+						.getString("context");
+				int userCount = resultActionByBucketActionContextCumulative
+						.getInt("user_count");
+				int distinctUserCount = resultActionByBucketActionContextCumulative
+						.getInt("distinct_user_count");
+
+				rollUpQuery = "REPLACE INTO experiment_rollup "
+						+ "(experiment_id, day, cumulative, bucket_label, action, "
+						+ "action_count, action_user_count,context) "
+						+ "values(unhex('" + experimentId + "'),'" + day
+						+ "',1,'" + bucket + "','" + action + "'," + userCount
+						+ "," + distinctUserCount + ",'" + context + "')";
+				insertIntoRollUp(rollUpQuery);
+			}
+		}
+	}
+
+	private void insertIntoRollUp(String rollUpQuery) throws SQLException {
+		Statement statement = connection.createStatement();
+
+		ResultSet resultSet = statement.executeQuery(rollUpQuery);
+
```

Comment:
Better have a try-with-resource here - if there are problems in the resultSet, the statement might not close this way.